packages/core/contracts/financial-templates/funding-rate-store/interfaces/FundingRateStoreInterface.sol

```diff
@@ -16,5 +16,11 @@ interface FundingRateStoreInterface {
      * @return FixedPoint.Signed representing the funding rate for the given contract. 0.01 would represent a funding
      * rate of 1% per second. -0.01 would represent a negative funding rate of -1% per second.
      */
-    function getFundingRateForContract(address perpetual) external view returns (FixedPoint.Signed memory);
+    function getFundingRateForContract(address perpetual) external returns (FixedPoint.Signed memory);
```

Comment:
noticed the `view` keyword is gone, how do you know when to use it or not?

Reply:
`view` means that the function does not modify contract state. Now that this method has the `publishAndWithdrawProposal` modifier, it can now change the `fundingRecords` state variable and transfer collateral tokens.